### DISCUSSION TITLE
Name change

### DISCUSSION THREAD

**Discussion by wilsonk** [david-vanderson/dvui#19] [Ideas]
Hello David, I saw in an earlier closed issue where another user had suggested a name change. I agree with that sentiment, as 'gui' is just too general and difficult to search for. I like their suggestions of 'dvui' or 'dvgui', but would also like to say 'zguid' (Zig GUI by David...pronounced like 'squid' with a 'z' instead of the 's' at the start), might be an option.

I also made up a quick mascot picture for 'zguid', just on a lark ;)

![zguid](https://github.com/david-vanderson/gui/assets/38802/70f30b66-9a8f-43c1-b763-c6bd56b419fd)

Again, you don't have to use this stuff...just spitballing ideas. Perhaps others have some good suggestions?

**Comment by wilsonk** [comment]
Oh, I should have said 'made up a quick mascot picture using AI', as I am not an artist and that was indeed, generated with AI ;)

**Comment by david-vanderson** [comment]
Awesome!  The more I think about it, I should probably find a decent 2-3 letter name but it doesn't have to have "ui" in it.  Does that make sense?

**Comment by wilsonk** [comment]
Not sure if you can capture the 'ui' part and and the Zig part of this project, while still being fairly unique, in 2-3 letters. Like 'zui' is already being used for a couple projects ('zgui' is also being used for a zig gui lib already...which is why I added the 'd' to the end...and the 'squid' play on words can make things easy to remember also). I think 'dvui' is pretty unique...and one less letter, so maybe that is the way to go. (There is a dv-ui but it is defunct at this point and isn't exactly the same). 

And you don't want to conflate this projects name with something pretty popular. A really good D language library called 'dlib' is constantly conflated with a great C++ lib of the same name. Makes it a pain to search for and get the correct results, specific to the language you are using!

**Comment by david-vanderson** [comment]
I agree.  I was thinking of having the name be something longer (like "ZigImmediateGui") but the import name be short (like `const zi = @import("ZigImmediateGui")`).  Does that make sense?

**Comment by wilsonk** [comment]
Yeah, that could work. Or maybe ZigImmediateMode...then the import can match the capitalization like ```const zim = ...```, which sort of looks nice and usable.

Otherwise, with ZigImmediateGui and matching the capitalization, you get ```const zig = ...``` and that won't work! Or one could try ZigImmediateModeGui which gives us ```const zimg = ...```, but then people might equate that with Zig image, I would think.

A longer name like this might work fine. Seems not bad.  Just a couple points to consider.

**Comment by david-vanderson** [comment]
I like that quite a bit.  The readme would say the name was "ZigImmediateMode GUI" usually imported as `zim`.

@iacore what do you think?

**Reply by iacore** [reply]
Assuming that the application only uses 1 UI library, isn't `ui` or `zui` better? `zim` seems like `vim`.

If imgui = Dear ImGui
Then ZigImmediateMode GUI = Zig ImGui ?

It might be mistaken as Zig bindings of imgui.

**Reply by david-vanderson** [reply]
Yes good points.  Naming is hard.  Maybe your original idea of `dvui` is best.  That at least doesn't conflict too much with other things and can't be confused for imgui bindings.  Feels a bit weird to use my initials.

I think in many cases this will be the only UI library in an application, but I hope to equally support bolting it on to an existing application (who knows if that will actually be used?).

**Comment by wilsonk** [comment]
Yep, naming is hard. I am leaning towards 'dvui' or 'zguid', but that is just my two cents. I wouldn't worry about using your initials...I have seen that enough in the past, that I doubt it would be seen negatively, or anything.

**Comment by david-vanderson** [comment]
I've pushed the change for dvui and will see how that plays out.  Thanks very much!

**Reply by iacore** [reply]
you need to change the repo name as well

**Reply by david-vanderson** [reply]
Done - thanks!
